server/src/kernels.cu

```diff
@@ -313,27 +473,33 @@ void diff::cuda::CUDACore::exec_core(uint8_t *frameData, uint8_t *showReadyNData
 
     // cudaMemsetAsync(d_pos, 0, sizeof *d_pos);
 
-    // Copy in the current pointer and run 
+    // Copy in the current pointer and run
 
-    #ifdef NOISE_FILTER
+#ifdef NOISE_FILTER
+    cudaMemcpyAsync(d_filtered, frameData, total, cudaMemcpyHostToDevice);
+    convolution_kernel<<<CDIM3(pgridSize), CDIM3(pblockSize)>>>(d_filtered, d_current);
+#else
+    cudaMemcpyAsync(d_current, frameData, total, cudaMemcpyHostToDevice);
+#endif
+#ifdef NOISE_FILTER
     CUDA_CHECK(cudaMemcpyAsync(d_filtered, frameData, total, cudaMemcpyHostToDevice));
     convolution_kernel<<<CDIM3(pgridSize), CDIM3(pblockSize)>>>(d_filtered, d_current);
-    #else
+#else
     CUDA_CHECK(cudaMemcpyAsync(d_current, frameData, total, cudaMemcpyHostToDevice));
-    #endif
-
-    #ifdef NOISE_VISUALIZER
-        #if NOISE_VISUALIZER == 1         
-        heat_map<<<1, nMaxThreads, 0>>>(d_current, d_previous, max4, d_noise_visualization);
-        CUDA_CHECK(cudaMemcpyAsync(showReadyNData, d_noise_visualization, total, cudaMemcpyDeviceToHost));
-        #elif NOISE_VISUALIZER == 2
-        red_black_map<<<1, nMaxThreads, 0>>>(d_current, d_previous, max4, d_noise_visualization);
-        CUDA_CHECK(cudaMemcpyAsync(showReadyNData, d_noise_visualization, total, cudaMemcpyDeviceToHost));
-        #endif
-    #endif
+#endif
```

Comment:
Is this duplicated?

Reply:
Let me check

Reply:
Solved https://github.com/MatteoBattilana/CUDAVideoStream/pull/4/commits/28d352251f33b667fb3f3c61601069de427f55c1